src/maker/api.rs

```diff
@@ -615,15 +613,35 @@ pub(crate) fn restore_broadcasted_contracts_on_reboot(maker: Arc<Maker>) -> Resu
         let next_internal_address = &maker.wallet.read()?.get_next_internal_addresses(1)?[0];
         let time_lock_spend = og_sc.create_timelock_spend(next_internal_address)?;
 
-        let tx = og_sc.get_fully_signed_contract_tx()?;
+        let tx = match og_sc.get_fully_signed_contract_tx() {
+            Ok(tx) => tx,
+            Err(ProtocolError::General(msg)) => {
+                log::error!("SOMETHING WENT WRONG: {}", msg);
+                continue;
```

Comment:
If we just continue it will get stuck in the same loop. We need to remove the swapcoin from the wallet.

Also log something like "Please open an issue reporting this with logs at <github issues link>".

Reply:
I think its doing this because you still have that og-swapcoin in the wallet, which is marked as unfinished and doesn't have signature data. So it will always catch that and try to recover, and then shoot this log.